### PR TITLE
Panel errors

### LINKED ISSUE #14528: React Panels & Error handling
* Handle and show query errors
* Handle and show panel visualization rendering errors.

### PR DESCRIPTION
**Pull request by jschill**
This handles
* Query errors
* Panel visualization rendering errors

Fixes #14528

### CODE REVIEW

File: public/app/core/components/Tooltip/Popper.tsx [hunk 1]
```diff
@@ -21,13 +21,21 @@ interface Props {
...
+
+export enum Themes {
+  Default = 'popper__background--default',
+  Error = 'popper__background--error',
 }
 
 class Popper extends PureComponent<Props> {
   render() {
-    const { children, renderContent, show, placement, refClassName } = this.props;
+    const { children, renderContent, show, placement, refClassName, theme } = this.props;
```
Comment (dprokop) [inline public/app/core/components/Tooltip/Popper.tsx]: @jschill please take a look at my changes in #14766. We will have to sync those.

File: public/app/core/components/Tooltip/withPopper.tsx [hunk 2]
```diff
@@ -9,13 +9,15 @@ export interface UsingPopperProps {
   content: string | ((props: any) => JSX.Element);
   className?: string;
   refClassName?: string;
+  theme?: string;
```
Comment (dprokop) [inline public/app/core/components/Tooltip/withPopper.tsx]: What do you think about specifying the type more precisely? `type TooltipTheme: 'default' | 'error' ...`

File: public/app/core/components/ErrorBoundary/ErrorBoundary.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+interface State {
+  error: Error;
+  errorInfo: ErrorInfo;
+}
+
+class ErrorBoundary extends Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = { error: null, errorInfo: null };
```
Comment (dprokop) [inline public/app/core/components/ErrorBoundary/ErrorBoundary.tsx]: As the state does not rely on props, we could move it to the class level and get rid of the constructor:

```ts
readonly state: State = {
  error: null;
  errorInfo: null
}
```

File: public/app/core/components/ErrorBoundary/ErrorBoundary.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    this.setState({
+      error: error,
+      errorInfo: errorInfo
+    });
+  }
+
+  render() {
+    const { error, errorInfo } = this.state;
+    return (
+      <>
```
Comment (dprokop) [inline public/app/core/components/ErrorBoundary/ErrorBoundary.tsx]: I think you can get rid of the fragment here, and return the children call.

File: public/app/plugins/panel/graph2/GraphPanel.tsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ import { Options } from './types';
 interface Props extends PanelProps<Options> {}
 
 export class GraphPanel extends PureComponent<Props> {
-  constructor(props) {
+  constructor(props: Props) {
```
Comment (dprokop) [inline public/app/plugins/panel/graph2/GraphPanel.tsx]: I think we can get rid of this constructor as React will set props anyway.

File: public/app/core/components/Tooltip/Popper.tsx (hunk 1 above)
Comment (jschill) [reply public/app/core/components/Tooltip/Popper.tsx]: Definately 👍

File: public/app/core/components/Tooltip/withPopper.tsx (hunk 2 above)
Comment (jschill) [reply public/app/core/components/Tooltip/withPopper.tsx]: Absolutely!

File: public/app/plugins/panel/graph2/GraphPanel.tsx (hunk 5 above)
Comment (jschill) [reply public/app/plugins/panel/graph2/GraphPanel.tsx]: 💯 right!

File: public/app/core/components/ErrorBoundary/ErrorBoundary.tsx (hunk 4 above)
Comment (jschill) [reply public/app/core/components/ErrorBoundary/ErrorBoundary.tsx]: Sweet!

File: public/app/core/components/ErrorBoundary/ErrorBoundary.tsx (hunk 3 above)
Comment (jschill) [reply public/app/core/components/ErrorBoundary/ErrorBoundary.tsx]: Nice!

**Pull request merged**